apps/emqx_connector/src/mqtt/emqx_connector_mqtt_schema.erl

```diff
@@ -294,5 +304,15 @@ desc(_) ->
 qos() ->
     hoconsc:union([emqx_schema:qos(), binary()]).
 
+downgrade_ingress_qos(QoS = 2) ->
```

Comment:
we can't log here, it's called in may different places.
i'm afraid we'll need to keep allowing QoS 2 from config, but the downgrade will need to done (and be logged) in where it is used.